src/Infrastructure/IndexManagement/ElasticIndexConfigurationRepository.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JeroenG\Explorer\Infrastructure\IndexManagement;
+
+use JeroenG\Explorer\Application\Explored;
+use JeroenG\Explorer\Application\IndexSettings;
+use JeroenG\Explorer\Domain\IndexManagement\IndexConfiguration;
+use JeroenG\Explorer\Domain\IndexManagement\IndexConfigurationRepository;
+use RuntimeException;
+
+class ElasticIndexConfigurationRepository implements IndexConfigurationRepository
```

Comment:
See previous question about finality.

Reply:
Final in this case will mean nobody can 'extend' and inject there own IndexConfigurationRepository based on ElasticIndexConfigurationRepository, I would advice against it.